File: Problemset/1704-Determine if String Halves Are Alike/Solution.rb
# @param {String} s
# @return {Boolean}
def halves_are_alike(s)
  count = 0

  (0...s.size).each do |i|
    count += i < s.size / 2 ? 1 : -1 if 'aeiouAEIOU'.include?(s[i])
  end

  count == 0
end

File: Problemset/1704-Determine if String Halves Are Alike/Solution.rs
impl Solution {
    pub fn halves_are_alike(s: String) -> bool {
        s.bytes()
            .enumerate()
            .filter(|(_, c)| b"aeiouAEIOU".contains(c))
            .map(|(i, _)| if i < s.len() / 2 { 1 } else { -1 })
            .sum::<i32>()
            == 0
    }
}
